samples/wifi/sta/src/main.c

```diff
@@ -317,3 +305,82 @@ int main(void)
 
 	return 0;
 }
+
+void start_wifi_thread(void);
+#define THREAD_PRIORITY K_PRIO_COOP(CONFIG_NUM_COOP_PRIORITIES - 1)
+K_THREAD_DEFINE(start_wifi_thread_id, CONFIG_STA_SAMPLE_STACK_SIZE,
+		start_wifi_thread, NULL, NULL, NULL,
+		THREAD_PRIORITY, 0, -1);
+
+void start_wifi_thread(void)
+{
+	if (k_sem_take(&wifi_ready_sem, K_FOREVER) == 0) {
+		if (wifi_status) {
+			start_app();
+		}
+	}
+}
+
+void wifi_ready_cb(bool wifi_ready)
+{
+	k_sem_reset(&wifi_ready_sem);
```

Comment:
This will abort any waiting semaphores with `-EAGAIN`, see https://docs.zephyrproject.org/latest/kernel/services/synchronization/semaphores.html#c.k_sem_reset. So, when you resume the thread, the behaviour might be undefined as you do both `k_sem_reset` + `k_sem_give`.